### PR TITLE
[Federation] Recompute registry name and version tag in the federation up script.sh and a bunch of other fixes.

### PR DESCRIPTION
**Pull request by madhusudancs**

### CODE REVIEW

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
Assigning this to @shashidharatd for review. Since I cannot assign to him in the "Assignees" drop down yet, putting my name there.

@shashidharatd PTAL.

File: federation/cluster/common.sh [hunk 1]
```diff
@@ -203,12 +203,14 @@ function create-federation-api-objects {
     export FEDERATION_ADMISSION_CONTROL="${FEDERATION_ADMISSION_CONTROL:-NamespaceLifecycle}"
 
     for file in federation-etcd-pvc.yaml federation-apiserver-{deployment,secrets}.yaml federation-controller-manager-deployment.yaml; do
+      echo "Creating manifest: ${file}"
+      $template "${manifests_root}/${file}"
       $template "${manifests_root}/${file}" | $host_kubectl create -f -
```
Comment (shashidharatd) [inline federation/cluster/common.sh]: Could have used tee command here

File: federation/cluster/common.sh [hunk 2]
```diff
@@ -203,12 +203,14 @@ function create-federation-api-objects {
...
     done
 
     # Update the users kubeconfig to include federation-apiserver credentials.
     CONTEXT=${FEDERATION_KUBE_CONTEXT} \
-	   KUBE_BEARER_TOKEN="$FEDERATION_API_TOKEN" \
+	   KUBE_BEARER_TOKEN="${FEDERATION_API_TOKEN}" \
```
Comment (shashidharatd) [inline federation/cluster/common.sh]: silly bug, good catch

File: federation/cluster/federation-up.sh [hunk 3]
```diff
@@ -84,6 +82,6 @@ if [[ "${USE_KUBEFED}" == "true" ]]; then
   init
   create_cluster_secrets
 else
-  export FEDERATION_IMAGE_TAG="$(get_version)"
+  export FEDERATION_IMAGE_TAG="$(echo ${KUBERNETES_RELEASE:-} | tr + _)"
```
Comment (shashidharatd) [inline federation/cluster/federation-up.sh]: so we had to export KUBERNETES_RELEASE to bring up federation outside CI environment?

**Comment by shashidharatd** [comment]
Thanks @madhusudancs, name of the PR is misleading, other than that LGTM

File: federation/cluster/common.sh (hunk 1 above)
Comment (shashidharatd) [inline federation/cluster/common.sh]: please ignore this comment

File: federation/cluster/federation-up.sh (hunk 3 above)
Comment (madhusudancs) [inline federation/cluster/federation-up.sh]: Yeah!

**Review by madhusudancs — COMMENTED** [review]
@shashidharatd changed the PR title and reorganized the commits a bit by squashing unnecessary changes. No other major changes. So I will add the LGTM label as soon as the tests pass.

Thanks for the review.

**Comment by madhusudancs** [comment]
@k8s-bot federation gce e2e test this

**Comment by madhusudancs** [comment]
Federation PR tests failed. I have looked at the errors. I will still push through this PR. Getting the tests to run is better than the state we are in right now.

**Comment by madhusudancs** [comment]
Adding the LGTM label.

**Pull request merged**